src/DF950.c

```diff
@@ -91,21 +273,164 @@ s32 func_80149828(void) {
     return D_80159AD4;
 }
 
-INCLUDE_ASM(s32, "DF950", sfx_stop_env_sounds);
+void sfx_stop_env_sounds(void) {
+    s32 i;
+    SoundInstance* sound;
+
+    if (!gGameStatusPtr->isBattle) {
+        D_80159AD0 = D_801598A0;
+        
+    } else {
+        D_80159AD0 = D_801599B8;
+    }
+
+    sound = D_80159AD0;
+    for (i = 0; i < MAX_SOUND_INSTANCES; i++) {
```

Comment:
I'll stop leaving these loop comments from here on, but basically, see if you can avoid incrementing a sound temp and do just arr[i]. also ,if you do need to increment the sound temp, try doing i++, sound++